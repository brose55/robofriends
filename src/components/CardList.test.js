import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList.js';

it('expect to render Card component', () => {
	const mockRobots = [
		{
			id: 0,
			name: 'Acid Burn',
			email: 'acidburn@hackers.com'
		},
		{
			id: 1,
			name: 'Crash Override',
			email: 'zerocool@hackers.com'
		}
	];
	expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});
