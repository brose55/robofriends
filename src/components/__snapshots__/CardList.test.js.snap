// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
      Array [
        Object {
          "email": "acidburn@hackers.com",
          "id": 0,
          "name": "Acid Burn",
        },
        Object {
          "email": "zerocool@hackers.com",
          "id": 1,
          "name": "Crash Override",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="acidburn@hackers.com"
          id={0}
          name="Acid Burn"
        />,
        <Card
          email="zerocool@hackers.com"
          id={1}
          name="Crash Override"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "email": "acidburn@hackers.com",
          "id": 0,
          "name": "Acid Burn",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "zerocool@hackers.com",
          "id": 1,
          "name": "Crash Override",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="acidburn@hackers.com"
            id={0}
            name="Acid Burn"
          />,
          <Card
            email="zerocool@hackers.com"
            id={1}
            name="Crash Override"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "email": "acidburn@hackers.com",
            "id": 0,
            "name": "Acid Burn",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "zerocool@hackers.com",
            "id": 1,
            "name": "Crash Override",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
